### DISCUSSION TITLE
rule that all structures with more than 8 fields are always passed as reference in methods should be removed

### DISCUSSION THREAD

**Discussion by i582** [vlang/v#18635] [Ideas]
### Describe the feature

This rule sounds as strange as possible, and what is most terrible, it is IMPLICIT and NOT DESCRIBED in the documentation.

I don't see the point in this rule because the developer can decide YOURSELF when the structure should be passed as reference or not.
Moreover, adding a new field to the structure completely changes the receiver's pass logic, which can be dangerous.

The compiler SHOULD NOT show off one's "intelligence" and add IMPLICITIES.

Example:

https://play.vlang.io/?query=06bda73b49 (8 fields)
```
7ffd53de0f90
7ffd53de0f30
```
All ok, different pointers because the receiver doesn't have a pointer type

https://play.vlang.io/p/88dc100004 (9 fields)
```
7ffe6b15c1ac
7ffe6b15c1ac
```
Not ok, same pointers.

Compiler code: https://github.com/vlang/v/blob/865c0ea8bd3e24704953c5633453d25d0f38d97f/vlib/v/parser/fn.v#L646

### Use Case

-

### Proposed Solution

Remove this rule.

### Acknowledgements

- [ ] I may be able to implement this feature request
- [ ] This feature might incur a breaking change

### Version used

V 0.3.2 89aa695.865c0ea

### Environment details (OS name and version, etc.)

```
OS: macos, macOS, 13.0.1, 22A400
Processor: 10 cpus, 64bit, little endian, Apple M1 Pro
CC version: Apple clang version 14.0.0 (clang-1400.0.29.202)

getwd: /Users/petrmakhnev/vlang-foundation/specification
vmodules: /Users/petrmakhnev/.vmodules
vroot: /Users/petrmakhnev/v
vexe: /Users/petrmakhnev/v/v
vexe mtime: 2023-01-22 21:48:01
is vroot writable: true
is vmodules writable: true
V full version: V 0.3.2 89aa695.865c0ea

Git version: git version 2.37.1 (Apple Git-137.1)
Git vroot status: weekly.2022.50-266-g865c0ea8-dirty
.git/config present: true
thirdparty/tcc status: thirdparty-macos-arm64 173c526e
```

**Comment by spytheman** [comment]
I agree 💯 - that is an anti feature, or even a bug imho.

**Comment by yuyi98** [comment]
Can the receiver of the method have only two modes, fn (mut f Foo) and fn (f Foo), and the second can only be passing by reference mode, canceling passing by value mode?
It's easier to use.

**Reply by JalonSolov** [reply]
It should be the opposite... `fn (mut f Foo)` would pass by reference, as it values should be modifiable.  `fn (f Foo)` would be pass by value, as they won't be modifiable.
